app/src/test/java/com/concordium/wallet/data/util/ParamsDeserializerUnitTest.kt

```diff
@@ -0,0 +1,38 @@
+package com.concordium.wallet.data.util
+import com.concordium.wallet.data.walletconnect.Params
+import com.concordium.wallet.data.walletconnect.ParamsDeserializer
+import com.google.gson.GsonBuilder
+import org.junit.Assert
+import org.junit.Test
+
+
+class ParamsDeserializerUnitTest {
+
+    @Test
+    fun testParamsDeserializer() {
```

Comment:
The name is too generic. The name should reflect what specifically is being tested. In example - DeserializerOutputShouldBeSimilarWhenSchemaIsPassedAsObjectAndString

Also we're missing all the test scenarios for when null is returned. And a test scenario to check that deserialization actually works. If the method just returns null regardless of the input, this test will pass.

Reply:
You could also follow the Should convention, and append _Should to the test class, and have each test method follow that as a sentence that can be read. 

In example.

Class name:  ParamsDeserializer_Should
Test function: Return_Same_Output_When_Same_Schema_Is_Passed_As_String_And_Object 

Then your test name should be written as such, so you can

Reply:
There is a class called ParamsDeserializer which methods are tested here and name reflects the naming style used in the other tests before, for example CurrencyUtilUnitTest for the CurrencyUtil., AccountDaoUnitTest for the AccountDao etc..
If null is returned, the test will fail with NullPointerException.

Reply:
Hi Sasho

The test name is too generic. It doesn't matter if it's consistently generic in other files. We should adhere to clean code principles, and have explicit names that express intent. Regardless of how it's done in different files, we need to understand what all functions (both test and non-test) do from their names. So please rename the function, and feel free to consider the options above or come up with something else, so it's very explicit what is being tested. 

I can't find any test covering that serialization is done correctly and which handles the different exception scenarios, such as JsonParseException, when deserialize function is called.

Null was a poor example, but if you modify the deserialize function to hardcore the output that is returned, no matter the input you pass to be deserialized, the unittest doesn't fail... Hence we need more tests, to test deserialization is done correctly by asserting/verifying the params.